services/src/main/java/com/chatbot/services/DialogflowFulfillmentController.java

```diff
@@ -0,0 +1,142 @@
+package com.chatbot.services;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import com.chatbot.services.protobuf.ChatServiceRequestOuterClass.ChatServiceRequest.ChatClient;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestHeader;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class DialogflowFulfillmentController {
+
+  @Autowired
+  private AsyncService asyncService;
+
+  private static final String SUGGEST_SERVICE_TIME_CHANGE = "SuggestServiceTimeChange";
+  private static final String CHANGE_SERVICE_TIME = "ChangeServiceTime";
+  private static final String PROVIDE_TARGETED_QUERIES_INTENT_NAME = "ProvideTargetedQueries";
+  private static final String RECOMMEND_MORE_OPTIONS_INTENT_NAME = "RecommendMoreOptions";
+  private static final String CHANGE_CATEGORY_INTENT_NAME = "ChangeCategory";
+  private static final String EXPLAIN_MEANING_INTENT_NAME = "ExplainMeaning";
+  private static final String UNRELATED_CALL_INTENT = "CallUnrelatedToBusinessCategory";
+  private static final String SUGGEST_TARGET_AREA_CHANGE = "SuggestTargetAreaChange";
+  private static final String CHANGE_TARGET_AREA = "ChangeTargetArea";
+  private static final String CALL_FROM_OUTSIDE_SERVICE_AREA_INTENT = 
+      "CallFromOutsideServiceArea";
```

Comment:
Would these be used anywhere outside this class? If so it may bee worth defining a separate constants file.

Reply:
updated